sim/core/attack.go

```diff
@@ -260,12 +263,76 @@ func (wa *WeaponAttack) setWeapon(weapon Weapon) {
 // inlineable stub for swing
 func (wa *WeaponAttack) trySwing(sim *Simulation) time.Duration {
 	if sim.CurrentTime < wa.swingAt {
+
+		if sim.Log != nil {
+			wa.unit.Log(sim, "trySwing failed - swing at %f and lastswing at %f", wa.swingAt, wa.lastSwingAt)
+		}
 		return wa.swingAt
 	}
 	return wa.swing(sim)
 }
 
+func (wa *WeaponAttack) castExtraAttacksStored(sim *Simulation) {
+	wa.castExtraAttacks(sim, &wa.extraAttacksStored, 0)
+	wa.extraAttacksStored = 0;
+
+	if wa.extraAttacksAura != nil {
+		wa.extraAttacksAura.SetStacks(sim, 0)
+	}
+}
+
+func (wa *WeaponAttack) castExtraAttacksTriggered(sim *Simulation, moreAttacks bool) {
+	if moreAttacks {
+		wa.castExtraAttacks(sim, &wa.extraAttacks, 0)
+	} else {
+		wa.castExtraAttacks(sim, &wa.extraAttacks, 1)
+	}
+}
+
+func (wa *WeaponAttack) castExtraAttacks(sim *Simulation, numExtraAttacks *int32, startIndex int32) bool {
```

Comment:
Did you intentionally make numExtraAttacks a pointer here?

Reply:
originally I needed to before using extraAttacksTemp and extraAttacksPending, but I can simplify it now

Reply:
simplified